packages/odf/components/create-storage-system/create-storage-system-steps/backing-storage-step/set-rbd-sc-default.tsx

```diff
@@ -0,0 +1,48 @@
+import * as React from 'react';
+import { useCustomTranslation } from '@odf/shared/useCustomTranslationHook';
+import { FormGroup, Checkbox } from '@patternfly/react-core';
+import { WizardDispatch, WizardState } from '../../reducer';
+import './backing-storage-step.scss';
+
+export const SetCephRBDStorageClassDefault: React.FC<SetCephRBDStorageClassDefaultProps> =
+  ({ dispatch, setCephRBDStorageClassDefault, doesDefaultSCAlreadyExists }) => {
+    const { t } = useCustomTranslation();
+
+    // for infra with already existing "default" SC (eg: say gp3-csi): option should be default unchecked.
+    // for infra with no "default" SC (BM here): option should be default checked.
+    React.useEffect(() => {
+      if (!doesDefaultSCAlreadyExists) {
+        dispatch({
+          type: 'backingStorage/setCephRBDStorageClassDefault',
+          payload: true,
+        });
+      }
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    return (
+      <FormGroup>
+        <Checkbox
+          id="set-rbd-sc-default"
+          label={t('Set Ceph RBD as the default StorageClass')}
+          description={t(
+            'Configure a default RBD StorageClass to eliminate manual annotations within a StorageClass or selecting a specific StorageClass when making storage requests or provisions in your PVCs.'
+          )}
+          isChecked={setCephRBDStorageClassDefault}
+          onChange={() =>
+            dispatch({
+              type: 'backingStorage/setCephRBDStorageClassDefault',
+              payload: !setCephRBDStorageClassDefault,
+            })
+          }
```

Comment:
```suggestion
          onChange={() =>
            setCephRBDStorageClassDefault(!doesDefaultSCAlreadyExists)
          }
```

useEffect will takecare of updating state

Reply:
that's diff use case, we only want `useEffect` to run only once...

Reply:
`setCephRBDStorageClassDefault` is a `boolean` value stored on wizard flow state...

Reply:
setCephRBDStorageClassDefault it not sound like boolean, it is like a function call. it is better to rename.

Reply:
I will rename...